src/abstractdataframe/abstractdataframe.jl

```diff
@@ -1035,7 +1035,12 @@ function _vcat(dfs::AbstractVector{<:AbstractDataFrame})
     length(header) == 0 && return DataFrame()
     cols = Vector{AbstractVector}(undef, length(header))
     for (i, name) in enumerate(header)
-        data = [df[name] for df in dfs]
+        # TODO: replace with commented out code after getindex deprecation
+        # data = [df[name] for df in dfs]
+        # the code below assumes that only DataFrame and SubDataFrame
+        # are subtypes of AbstractDataFrame
+        # it should be removed ASAP after deprecation
+        data = [df isa DataFrame ? df[name] : view(parent(df)[name], rows(df)) for df in dfs]
```

Comment:
It shouldn't matter much since it's temporary, but better call `view` only for `SubDataFrame`, so that the fallback works for any type. Then you don't need the scary comment.

Reply:
But I cannot do this here because `SubDataFrame` is not defined at this stage of DataFrames.jl loading. And I would have to move the whole function much later in the loading process to make it work, which I think should not be done since it is only a temporary change.

Reply:
Argh. Well, let's say we will remove this soon anyway...